src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Tests/Controllers/PublicationsControllerTests.cs

```diff
@@ -0,0 +1,207 @@
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Tests.Fixture;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Tests.Fixture.DataFixtures;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Api.ViewModels;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Model;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Model.Database;
+using GovUk.Education.ExploreEducationStatistics.Common.Tests.Extensions;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.AspNetCore.WebUtilities;
+using System.Net;
+using System.Net.Http.Json;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Tests.Services;
+
+namespace GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Tests.Controllers;
+
+public class PublicationsControllerTests : IntegrationTestFixture
+{
+    public PublicationsControllerTests(TestApplicationFactory testApp) : base(testApp)
+    {
+    }
+
+    [Theory]
+    [InlineData(1, 2, 2, 1)]
+    [InlineData(2, 2, 2, 1)]
+    [InlineData(1, 2, 2, 10)]
+    [InlineData(1, 2, 9, 1)]
+    [InlineData(2, 2, 9, 1)]
+    [InlineData(1, 2, 9, 10)]
+    [InlineData(1, 3, 2, 1)]
+    public async Task ListPublications_MultiplePublishedDataSets_Returns200WithAllPublicationsForPublishedDataSets(
+        int page,
+        int pageSize,
+        int numberOfPublishedDataSets,
+        int numberOfUnpublishedDataSets)
+    {
+        var client = SetupApp(new ContentApiClientMock()).CreateClient();
+
+        var publishedDataSets = GeneratePublishedDataSets(numberOfPublishedDataSets);
+        var unpublishedDataSets = GenerateUnublishedDataSets(numberOfUnpublishedDataSets);
+
+        var allDataSets = publishedDataSets.Concat(unpublishedDataSets);
+
+        await AddTestDataSets(allDataSets.ToArray());
+
+        var response = await ListPublications(client, page, pageSize);
+
+        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+        var content = await response.Content.ReadFromJsonAsync<PaginatedPublicationListViewModel>();
```

Comment:
I'm not a fan of adding a `Newtonsoft.Json` attribute to the `PaginatedPublicationListViewModel` type, because then I'd be introducing Newtonsoft into this project.

I also don't think we should just go back to `Newtonsoft.Json` just for this.

I think for me, the 2 solutions I'd prefer, out of the ones you suggested, are:
- Don't use `HttpResponseMessageTestExtensions` and just leave the test as it is
- Use a new method like `AssertOkSystemJson` so that eventually there will be zero references to the old method and we'd know we can delete it. If we used a flag (which defaults to newtsonsoft), that would have the disadvantage of everything still referencing the same method. But I don't think there is much preference between these for me.

A 3rd solution would be to use reflection to read the attributes decorating the type. And the single method that we (`AssertOk`) already have can handle both cases? But it's a bit messy.

Reply:
Given this a little thought and you're probably right. Going back to Newtonsoft probably isn't the right call.

Your third solution is an interesting one, but the code is probably complex and not very nice. For example, we'd need to recursively go through the entire object graph looking for these attributes. It's a lot of work.

That leaves the first two solutions :thinking: 

Personally, it'd be good to use the extensions to keep the test code style consistent across the board.

Perhaps we go with the approach of adding new methods. Whilst we could use a parameter flag, it would mean having to make changes to any existing usages of `AssertOk`, which probably isn't too desirable at this point.

Reply:
If we use an optional parameter flag with a default value we won't need to change the other usages of `AssertOk`.

Reply:
went with the additional optional parameter :)